mantidimaging/core/utility/unit_conversion.py

```diff
@@ -11,18 +11,42 @@ class UnitConversion:
     planck_h: float = 6.62606896e-34  # [JHz-1]
     angstrom: float = 1e-10  # [m]
     mega_electro_volt: float = 1.60217662e-19 / 1e6
+    target_to_camera_dist: float = 56  # [m]
+    data_offset: float = 0  # [us]
+    tof_data_to_convert: np.ndarray
+    velocity: np.ndarray
 
-    def __init__(self, data_to_convert: np.ndarray, target_to_camera_dist: float = 56) -> None:
-        self.tof_data_to_convert = data_to_convert
-        self.target_to_camera_dist = target_to_camera_dist
-        self.velocity = self.target_to_camera_dist / self.tof_data_to_convert
+    def __init__(self, data_to_convert: np.ndarray | None = None) -> None:
+        if data_to_convert is not None:
+            self.set_data_to_convert(data_to_convert)
 
     def tof_seconds_to_wavelength(self) -> np.ndarray:
+        self.check_data()
         wavelength = self.planck_h / (self.neutron_mass * self.velocity)
         wavelength_angstroms = wavelength / self.angstrom
         return wavelength_angstroms
 
     def tof_seconds_to_energy(self) -> np.ndarray:
+        self.check_data()
         energy = self.neutron_mass * self.velocity / 2
         energy_evs = energy / self.mega_electro_volt
         return energy_evs
+
+    def tof_seconds_to_us(self) -> np.ndarray:
+        self.check_data()
+        return (self.tof_data_to_convert + self.data_offset) * 1e6
+
+    def set_target_to_camera_dist(self, target_to_camera_dist: float) -> None:
+        self.target_to_camera_dist = target_to_camera_dist
```

Comment:
Given that the method does not do anything fancy, I just let users of class set the value directly. So the presenter.py below would do
```
self.model.units.target_to_camera_dist = self.view.flightPathSpinBox.value()
```

Reply:
I left it generalised like that just in case something like that would be required in the future but for now we could just set the attribute directly

Reply:
We can always turn it into a property in the future if needed. (This is a python specific difference from the refactoring book).